File: charcutaria-business-manager/src/app/shared/services/corpclient/corp-client.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { CorpClient } from '../../models/corpClient';
import { environment } from 'src/environments/environment';
import { Observable, of } from 'rxjs';
import { ClientRegistration } from '../../models/clientRegistration';
import { flatMap } from 'rxjs/operators';

@Injectable({
  providedIn: 'root'
})
export class CorpClientService {


  constructor(private httpClient: HttpClient) { }
  public GetCorpClient(corpClientId: number): Observable<CorpClient> {
    return this.httpClient.get<CorpClient>(`${environment.apiUrl}/CorpClient/${corpClientId}`)
      .pipe(flatMap(r => {
        r.socialIdentifier = r.customerTypeId == 1 ? r.cpf : r.cnpj;
        return of(r);
      }));
  }

  public GetPaged(page: number, pageSize: number, filter: string, active: boolean | null): Observable<CorpClient[]> {
    let params = new HttpParams();
    if (filter)
      params = params.append('filter', filter);
    if (active != null)
      params = params.append('active', String(active));
    return this.httpClient.get<CorpClient[]>(`${environment.apiUrl}/CorpClient/${page}/${pageSize}`, { params: params });
  }
  public Register(registrationData: ClientRegistration): Observable<CorpClient> {
    return this.httpClient.post<CorpClient>(`${environment.apiUrl}/CorpClient/Register`, registrationData);
  }
  public Update(corpClient: CorpClient): Observable<CorpClient> {
    return this.httpClient.put<CorpClient>(`${environment.apiUrl}/CorpClient`, corpClient);
  }
  public GetActives(): Observable<CorpClient[]> {
    return this.httpClient.get<CorpClient[]>(`${environment.apiUrl}/CorpClient/Actives`);
  }
}

File: Charcutarie.Api/Charcutarie.Services/CorpClientService.cs
using Charcutarie.Application.Contracts;
using Charcutarie.Core.ExceptionHandling;
using Charcutarie.Core.SMTP;
using Charcutarie.Models;
using Charcutarie.Models.ViewModels;
using Charcutarie.Services.Contracts;
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Transactions;

namespace Charcutarie.Services
{
    public class CorpClientService : ICorpClientService
    {
        private readonly ICorpClientApp corpClientApp;
        private readonly IUserApp userApp;
        private readonly IEmailManager emailManager;

        public CorpClientService(ICorpClientApp corpClientApp, IUserApp userApp, IEmailManager emailManager)
        {
            this.corpClientApp = corpClientApp;
            this.userApp = userApp;
            this.emailManager = emailManager;
        }
        public async Task<PagedResult<CorpClient>> GetPaged(int page, int pageSize, string filter, bool? active = null)
        {
            return await corpClientApp.GetPaged(page, pageSize, filter, active);
        }
        public async Task<CorpClient> Register(ClientRegistration model)
        {
            model.Mobile = $"+55 {model.Mobile}";
            var type = model.CustomerTypeId == 1 ? "CPF" : "CNPJ";
            var existingClient = await (model.CustomerTypeId == 1 ?
                    corpClientApp.GetByCpf(model.SocialIdentifier) :
                    corpClientApp.GetByCnpj(model.SocialIdentifier));
            if (existingClient != null)
                throw new BusinessException($"{type} já cadastrado no sistema.");

            var result = await corpClientApp.Add(new NewCorpClient
            {
                CNPJ = model.CustomerTypeId == 1 ? string.Empty : model.SocialIdentifier,
                CPF = model.CustomerTypeId == 2 ? string.Empty : model.SocialIdentifier,
                Currency = "R$",
                CustomerTypeId = model.CustomerTypeId,
                DBAName = model.CustomerTypeId == 1 ? model.Name : model.DbaName,
                Name = model.Name,
                LicenseExpirationDate = null,
                Active = true,
                Email = model.Email,
                Mobile = model.Mobile
            });
            var user = await userApp.Add(new NewUser
            {
                Active = true,
                DefaultMassMid = 2,
                DefaultVolumeMid = 6,
                Email = model.Email,
                Mobile = model.Mobile,
                Name = model.Name,
                Password = model.Password,
                RoleId = 2,
                Username = model.Username
            }, result.CorpClientId);
            emailManager.SendRegistrationEmail(result.AccountNumber, model.Username, model.Email, model.Name, model.CustomerTypeId, model.SocialIdentifier);
            return result;
        }

        public async Task<CorpClient> Update(UpdateCorpClient model)
        {
            return await corpClientApp.Update(model);
        }
        public async Task<CorpClient> Get(int id)
        {
            return await corpClientApp.Get(id);
        }

        public async Task<IEnumerable<CorpClient>> GetActives()
        {
            return await corpClientApp.GetActives();
        }
    }
}
